### PR TITLE
Fix bug with date parsing month and day

### PR DESCRIPTION
**Pull request by greyhillman**
It was returning a date in the future instead of in the past.
For example, "July 20" would return "2018-07-20" instead of "2017-07-20"
when the current year is 2017.

Doing `friends add activity July 20` would result in
`2018-07-20: `
when it should have been:
`2017-07-20: `

---

Hi there! Thanks so much for submitting a pull request!

Let's just make sure together that all of these boxes are checked before we
merge this change:

- [x] The code in these changes works correctly.
- [ ] Code has tests as appropriate.
- [ ] Code has been reviewed by @JacobEvelyn.
- [x] All tests pass on Travis CI.
- [x] Rubocop reports no issues on Travis CI.
- [x] The `README.md` file is updated as appropriate.

Don't worry—this list will get checked off in no time!

### CODE REVIEW

**Comment by JacobEvelyn** [comment]
Thanks so much for submitting this PR! Sorry I didn't see it yesterday.

File: lib/friends/activity.rb [hunk 1]
```diff
@@ -34,9 +34,15 @@ def initialize(str: "")
...
 
-      # rubocop:disable Lint/AssignmentInCondition
-      if time = (date_s =~ /^\d{4}-\d{2}-\d{2}$/ ? Time : Chronic).parse(date_s)
-        # rubocop:enable Lint/AssignmentInCondition
+      time = if date_s =~ /^\d{4}-\d{2}-\d{2}$/
+               Time.parse(date_s)
+             else
+               # If the user inputed a non YYYY-MM-DD format, asssume
+               # it is in the past
+               Chronic.parse(date_s, { context: :past })
```
Comment (JacobEvelyn) [inline lib/friends/activity.rb]: Not a big deal, but it looks like you don't need the hashes here. This can just be:

```
Chronic.parse(date_s, context: :past)
```

File: test/commands/add/activity_spec.rb [hunk 2]
```diff
@@ -29,12 +29,31 @@ def date_parsing_specs(test_stdout: true)
...
-    describe "when date is natural language" do
+    describe "when date is natural language and in full" do
       let(:date) { "February 23rd, 2017" }
 
       it { line_added "- 2017-02-23: #{description}" }
       it { stdout_only "Activity added: \"2017-02-23: #{description}\"" } if test_stdout
     end
+
+    describe "when date is natural language and only month and day" do
+      mocked_time = Time.new(2017, 8, 1) # First of August
```
Comment (JacobEvelyn) [inline test/commands/add/activity_spec.rb]: Would you mind changing the mocked year to 2016 or earlier? If the mocking doesn't work the way we think it does (or, for instance, Chronic's internals change and it stops using `Time.now`), I'd like to make sure that this test fails rather than passing because it happens to be 2017 now.

Does that make sense?

Comment (greyhillman) [inline test/commands/add/activity_spec.rb]: Yep, that makes sense.

File: lib/friends/activity.rb (hunk 1 above)
Comment (greyhillman) [inline lib/friends/activity.rb]: Don't know too much Ruby. I knew Rubocop was giving me a warning about something with hashes. Thanks!

Comment (JacobEvelyn) [inline lib/friends/activity.rb]: Ah, well your code looks really nice for someone new to Ruby! :1st_place_medal:

**Comment by greyhillman** [comment]
Do you know how to mock out `Time.now` correctly? I've been trying to change the time back to 2016 but the stub seems to not work. I've added an assertion to check if the stub works and it keeps failing.

**Comment by JacobEvelyn** [comment]
Ah! This one stumped me for quite a while too! There's a few things going on here:

1. Minitest is a bit unintuitive at first about what you can call where, since everything is nested blocks that execute at various times. (For instance, I think you need to call `stub` within the `it` block rather than around it.) I'm not positive, but I suspect your test code might not be executing the way it appears to be as a result.
2. More importantly, the tests for `friends` are a bit unusual in that each test is actually forking a new process and executing the command-line `friends` program and then inspecting its output and how it changed the `friends.md` file. This is nice in that it lets us test things much, much more easily/safely/effectively than if our tests were along the lines of traditional "run this method for this class" (I can say this confidently because this was what the first version of these tests was, and as codebase complexity grew the tests became very unwieldy and missed some tricky edge cases). *But* that means that stubbing will happen in the *test process* but not affect the *forked call-the-program process*, which I had completely forgotten about when I first looked at your code.
3. Most importantly, Chronic's `context: :past` behavior sadly does not work. 😭 At first I thought it was just the result of the published gem version being woefully out of date (which it is) since some old GitHub Issues had implied that `context: :past` had been "fixed" at some point. I even went so far as to [publish the latest version to RubyGems](https://rubygems.org/gems/chronic-master) and use that in a test branch but nope, no matter what I do it always seems to return dates in the future. I've created an [Issue](https://github.com/mojombo/chronic/issues/361) for this in the Chronic repo but since that project seems somewhat poorly maintained I also took a stab at a semi-hacky workaround.

You can see my version in [this PR](https://github.com/JacobEvelyn/friends/pull/181). I'm happy to merge that and get the fix out for you or wait and let you incorporate some of that into your own branch so you can get credit as a [contributor](https://github.com/JacobEvelyn/friends/graphs/contributors). Let me know what you'd prefer!

Also please let me know if I can explain anything more clearly!

**Comment by greyhillman** [comment]
Thanks for the explanation!

You can merge the new PR instead of this one. I just have some comments to make about the new PR though. I'll post them there.

**Comment by JacobEvelyn** [comment]
Awesome, thanks!

**Comment by JacobEvelyn** [comment]
This is now fixed in version 0.33. Thanks so so much for all the help, @greyhillman!
